src/doom/r_main.c

```diff
@@ -782,6 +783,15 @@ void R_Init (void)
     R_InitSkyMap ();
     R_InitTranslationTables ();
     printf (".");
+    // Moire error emulation
+    if(gameversion <= exe_doom_1_2)
+    {
+        R_SetHeighFracBits(16);
```

Comment:
Is there a reason why these numbers (16 and 12) are specified in this file instead of in `r_segs.c`? Given that there are only two possible values I don't see why you need a general function to set the number of bits. Probably a better approach would be:

    void R_SetHeightFracBits(void);

Then have that function initialize `heightbits` / `heightunit` to one of two constants depending on `gameversion`. ie. move the conditional code out of this file.